bin/diagnostics_functions.js

```diff
@@ -0,0 +1,57 @@
+const process = require('process');
+const util = require('./shared_utils');
+const config = require('./config').config;
+const request = require('superagent');
+const fs = require('fs');
+
+module.exports.dispatcher = function (args) {
+    if (args.length === 0) {
+        printModuleHelp();
+        process.exit(1);
+    }
+    switch (args[0].toLowerCase()) {
+        case 'run': {
+            let options = util.optionsFromArgs(args.splice(1), ['output', 'reportFileName']);
+            runDiagnostics(options['output'], options['reportFileName']);
+            break;
+        }
+        case 'help':
+            printModuleHelp();
+            break;
+        default:
+            util.printErrorAndExit("Unknown operation");
+    }
+}
+
+function runDiagnostics(outputFolder, fileName) {
+    let endPoint = config.server + '/api/v1/diagnostics';
```

Comment:
its good to use const instead of let if the variable's value is not changing. I see let in several other places in this file. please change them to const.